scripts/thief.bicep

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license information.
+
+param location string
+param prefix string
+param thief_runs_resource_group string
+param thief_app_insights_connection_string string
+param thief_container_registry_host string
+param thief_container_registry_password string
+param thief_container_registry_shortname string
+param thief_container_registry_user string
+param user_principal_id string
+param thief_shared_subscription_id string
+param thief_shared_keyvault_name string
+param thief_shared_resource_group string
+
+resource thief_iot_hub 'Microsoft.Devices/IotHubs@2020-08-01' = {
+  name: '${prefix}-thief-hub'
+  location: location
+  sku: {
+    name: 'S2'
+    capacity: 1
+  }
+  properties: {
+    eventHubEndpoints: {
+      events: {
+        retentionTimeInDays: 1
+        partitionCount: 4
+      }
+    }
+    routing: {
+      routes: [
+        {
+          name: 'twin-update-event'
+          source: 'TwinChangeEvents'
+          condition: 'true'
+          endpointNames: [
+            'events'
+          ]
+          isEnabled: true
+        }
+      ]
+      fallbackRoute: {
+        name: '$fallback'
+        source: 'DeviceMessages'
+        condition: 'true'
+        endpointNames: [
+          'events'
+        ]
+        isEnabled: true
+      }
+    }
+  }
+}
+var shared_access_key_name = '${listKeys(thief_iot_hub.id, '2020-04-01').value[0].keyName}'
+var shared_access_key = '${listKeys(thief_iot_hub.id, '2020-04-01').value[0].primaryKey}'
+var thief_iot_hub_connection_string = 'HostName=${thief_iot_hub.name}.azure-devices.net;SharedAccessKeyName=${shared_access_key_name};SharedAccessKey=${shared_access_key}'
+var thief_eventhub_connection_string = 'Endpoint=${thief_iot_hub.properties.eventHubEndpoints.events.endpoint};SharedAccessKeyName=${shared_access_key_name};SharedAccessKey=${shared_access_key};EntityPath=${thief_iot_hub.properties.eventHubEndpoints.events.path}'
+
+resource thief_dps 'Microsoft.Devices/provisioningServices@2020-03-01' = {
+  name: '${prefix}-thief-dps'
+  location: location
+  sku: {
+    name: 'S1'
+    capacity: 1
+  }
+  properties: {
+    state: 'Active'
+    provisioningState: 'Succeeded'
+    iotHubs: [
+      {
+        connectionString: thief_iot_hub_connection_string
+        location: location
+      }
+    ]
+    allocationPolicy: 'Hashed'
+  }
+}
+
+resource thief_key_vault 'Microsoft.KeyVault/vaults@2016-10-01' = {
+  name: '${prefix}-thief-kv'
+  location: location
+  properties: {
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: subscription().tenantId
+    accessPolicies: [
+      {
+        tenantId: subscription().tenantId
+        objectId: thief_container_identity.properties.principalId
+        permissions: {
+          secrets: [
+            'get'
+          ]
+        }
+      }
```

Comment:
should there be a comma at the end of this line? These look like objects in an array.

Reply:
I've not see a usage of comma even for arrays in bicep